src/util/splitTextByTerm/splitTextByTerm.spec.ts

```diff
@@ -0,0 +1,28 @@
+import { splitTextByTerm } from './splitTextByTerm'
+
+describe('splitTextByTerm', () => {
+  it('splits text by the search term', () => {
+    const result = splitTextByTerm('Hello world', 'world')
+    expect(result).toEqual(['Hello ', 'world'])
+  })
+
+  it('is case insensitive', () => {
+    const result = splitTextByTerm('Hello World', 'world')
+    expect(result).toEqual(['Hello ', 'World'])
+  })
+
+  it('handles multiple instances of the search term', () => {
+    const result = splitTextByTerm('worldworld world', 'world')
+    expect(result).toEqual(['world', 'world', ' ', 'world'])
+  })
+
+  it('returns the original text when the search term is not found', () => {
+    const result = splitTextByTerm('Hello world', 'test')
+    expect(result).toEqual(['Hello world'])
+  })
+
+  it('returns the original text when the search term is empty', () => {
+    const result = splitTextByTerm('Hello world', '')
+    expect(result).toEqual(['Hello world'])
+  })
+})
```

Comment:
Maybe you should add a test where the original text is empty too? It could also happen